RecipeApp.Core/RecipeConverter.cs

```diff
@@ -0,0 +1,270 @@
+
+            var cookingUnit = GetClosestCookingUnit(ml);
+            var multiplier = Conversion.CookingUnits[cookingUnit];
+            var convertedAmount = ml / multiplier;
+
+            words[index - 1] = convertedAmount.ToString("F2", CultureInfo.InvariantCulture);
+            words[index] = cookingUnit;
+        }
+
+        private static string ConvertToCookingUnits(string recipe)
+        {
+            string[] words = recipe.Split(' ', '\n');
+
+            for (int index = 0; index < words.Length; index++)
+            {
+                try
+                {
+                    ConvertToCookingUnit(index, words);
+                }
+                catch (InvalidRecipeException ex)
+                {
+                    Console.WriteLine($"Skipping word, because: {ex.Message}");
+                }
+            }
+
+            return string.Join(" ", words);
+        }
+
+        private static string GetClosestCookingUnit(double ml)
+        {
+            var smallestConversion = double.MaxValue;
+            // 2 is the smallest multiplier.
+            var closestCookingUnit = "gallon";
+
+            foreach (var unit in Conversion.CookingUnits)
+            {
+                var coefficient = unit.Value;
+
+                if (coefficient > ml) continue;
+
+                var conversion = ml / coefficient;
+                if (conversion < smallestConversion)
+                {
+                    smallestConversion = conversion;
+                    closestCookingUnit = unit.Key;
+                }
+            }
+
+            return closestCookingUnit;
+        }
+
+        private static double GetSiUnit(int index, string[] words)
+        {
+            if (words[index] == "ml" || words[index] == "l")
+            {
+                var isNumber = ParseDouble(words[index - 1], out var siAmount);
+                if (!isNumber) return -1;
+
+                return siAmount;
+            }
+            else
+            {
+                return -1;
+            }
+        }
+
+        private static BaseUnit GetUnit(int idx, string[] words)
+        {
+            string unitName = words[idx];
+
+            Type unitType = null;
+            foreach (var baseUnitType in Assembly.GetExecutingAssembly().GetTypes().Where(t => (t.BaseType?.Name ?? "") == "BaseUnit"))
+            {
+                var tempUnit = Assembly.GetExecutingAssembly().CreateInstance(baseUnitType.FullName) as BaseUnit;
+                var tempUnitName = baseUnitType.GetProperties().FirstOrDefault(f =>f.Name == "Name")?.GetValue(tempUnit).ToString() ?? "";
+                if ( tempUnitName == unitName || tempUnitName+ "s" == unitName)
+                {
+                    unitType = baseUnitType;
+                    break;
+                }
+            }
+            if (unitType is null) return null;
+
+            BaseUnit unit = null;
+            if (ParseDouble(words[idx - 1], out double amount))
```

Comment:
It's more readable if you move this to a separate line.

Reply:
reformatted the whole thing ;D